examples/service/rpc_client.py

```diff
@@ -1,25 +1,25 @@
 import asyncio
 from miteD.service.client import RemoteService
 from contextlib import suppress
+from nats.aio.client import Client as NatsClient
 
 
-async def main(_loop):
-    client_1_0 = RemoteService(name='test', version='1.0', loop=_loop)
-    client_1_1 = RemoteService(name='test', version='1.1', loop=_loop)
-    client_2_0 = RemoteService(name='test', version='2.0', loop=_loop)
+async def main(loop):
+    nc = NatsClient()
+    await nc.connect(io_loop=loop, servers=['nats://127.0.0.1:4222'], verbose=True)
+    client_1_0 = RemoteService(name='test', version='1.0', nc=nc)
+    client_1_1 = RemoteService(name='test', version='1.1', nc=nc)
+    client_2_0 = RemoteService(name='test', version='2.0', nc=nc)
     print(await client_1_0.ping())
-    print(await client_1_0.ping())
-    print(await client_2_0.ping())
     print(await client_1_1.ping())
+    print(await client_2_0.ping())
     print(await client_2_0.hello('world'))
     print(await client_2_0.add(12, 54))
 
 if __name__ == '__main__':
     loop = asyncio.get_event_loop()
     loop.run_until_complete(main(loop))
-    pending = asyncio.Task.all_tasks(loop)
-    for task in pending:
-        task.cancel()
-        with suppress(asyncio.CancelledError):
-            loop.run_until_complete(task)
+    group = asyncio.gather(*asyncio.Task.all_tasks(), return_exceptions=True)
+    group.cancel()
+    loop.run_until_complete(group)
```

Comment:
isn't the suppress context needed here?

Reply:
The 'return_exceptions=True' parameter means that any exception that occur will not be propagated up but returned with the 'group' object.

Reply:
Sorry overlooked that comment.
This is an idiomatic way of exiting async programs.
main(loop) may have spawned some tasks which can still be running even though main exited. It may not be obvious in some more complex code.

Reply:
Cool. Thanks for the response.